### PR TITLE
AML-3636 Migrate api/accounts

### PR DESCRIPTION
**Pull request by Howesy77**

### CODE REVIEW

File: src/SFA.DAS.EmployerAccounts.Web/Orchestrators/EmployerAccountPayeOrchestrator.cs [hunk 1]
```diff
@@ -170,7 +170,7 @@ public virtual async Task<OrchestratorResponse<RemovePayeSchemeViewModel>> GetRe
                     });
 
             var payeResponse = await
-                Mediator.SendAsync(new GetPayeSchemeByRefQuery
+                Mediator.SendAsync(new GetPagedEmployerAccountsQuery
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EmployerAccounts.Web/Orchestrators/EmployerAccountPayeOrchestrator.cs]: Why has this changed?  Will that even work?

File: src/SFA.DAS.EmployerApprenticeshipsService.Application/Services/EmployerAccountsApi/EmployerAccountsApiService.cs [hunk 2]
```diff
@@ -13,10 +15,16 @@ public class EmployerAccountsApiService : IEmployerAccountsApiService
...
+        private readonly IMapper _mapper;
 
-        public EmployerAccountsApiService(IEmployerAccountsApiHttpClientFactory employerAccountsApiHttpClientFactory, ILog log)
+        public EmployerAccountsApiService(
+            IEmployerAccountsApiHttpClientFactory employerAccountsApiHttpClientFactory, 
+            ILog log,
+            IMapper mapper)
         {
             _log = log;
+            _mapper = mapper;
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EmployerApprenticeshipsService.Application/Services/EmployerAccountsApi/EmployerAccountsApiService.cs]: Is the mapper actually being used?

File: src/SFA.DAS.EAS.Account.Api.UnitTests/Controllers/EmployerAccountsControllerTests/EmployerAccountsControllerTests.cs [hunk 3]
```diff
@@ -31,18 +32,21 @@ public void Arrange()
...
-            var orchestrator = new AccountsOrchestrator(Mediator.Object, Logger.Object, Mapper.Object, HashingService.Object);
+            ApiService = new Mock<IEmployerAccountsApiService>();
+
+            var orchestrator = new AccountsOrchestrator(Mediator.Object, Logger.Object, Mapper.Object, HashingService.Object, ApiService.Object);
             Controller = new EmployerAccountsController(orchestrator);
 
             UrlHelper = new Mock<UrlHelper>();
             Controller.Url = UrlHelper.Object;
-
+            /*
```
[the remark this answers is not in the document]
Comment (Howesy77) [reply src/SFA.DAS.EAS.Account.Api.UnitTests/Controllers/EmployerAccountsControllerTests/EmployerAccountsControllerTests.cs]: Done

File: src/SFA.DAS.EAS.Account.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs [hunk 4]
```diff
@@ -3,268 +3,46 @@
...
+using AccountWithBalanceViewModel = SFA.DAS.EAS.Account.Api.Types.AccountWithBalanceViewModel;
 
 namespace SFA.DAS.EAS.Account.Api.UnitTests.Controllers.EmployerAccountsControllerTests
 {
     [TestFixture]
     public class WhenIGetAccounts : EmployerAccountsControllerTests
     {
         [Test]
-        public async Task ThenAccountsAreReturnedWithTheirBalanceAndAUriToGetAccountDetails()
+        public async Task ThenShouldReturnOkNegotiatedContentResultWithStatistics()
```
[the remark this answers is not in the document]
Comment (Howesy77) [reply src/SFA.DAS.EAS.Account.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs]: No idea, reverted

File: src/SFA.DAS.EAS.Account.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs [hunk 5]
```diff
@@ -3,268 +3,46 @@
...
-
-            //Assert
-            var model = actual as OkNegotiatedContentResult<PagedApiResponseViewModel<AccountWithBalanceViewModel>>;
-            Assert.IsNotNull(model);
-            Assert.IsNotEmpty(model.Content.Data);
-            Assert.IsTrue(model.Content.Data.All(c => c.IsLevyPayer));
-        }
-
-        [Test]
-        public async Task ThenIfThereIsDataFromTheAccountBalanceQueryThenTheLevyOverFlagIsUsed()
```
[the remark this answers is not in the document]
Comment (Howesy77) [reply src/SFA.DAS.EAS.Account.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs]: Some moved to the EmployerAccount API as that control the paging etc
Some removed as they seem to be testing Levy status
Some seem irrelevant

File: src/SFA.DAS.EmployerAccounts.Api/Orchestrators/OrchestratorResponse.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Net;
+
+namespace SFA.DAS.EmployerAccounts.Api.Orchestrators
+{
+    public class OrchestratorResponse
```
[the remark this answers is not in the document]
Comment (Howesy77) [reply src/SFA.DAS.EmployerAccounts.Api/Orchestrators/OrchestratorResponse.cs]: Done

File: src/SFA.DAS.EmployerAccounts.Web/Orchestrators/EmployerAccountPayeOrchestrator.cs (hunk 1 above)
Comment (Howesy77) [reply src/SFA.DAS.EmployerAccounts.Web/Orchestrators/EmployerAccountPayeOrchestrator.cs]: No idea how this happened :)

File: src/SFA.DAS.EAS.Account.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs [hunk 7]
```diff
@@ -20,251 +21,28 @@ public class WhenIGetAccounts : EmployerAccountsControllerTests
...
-                    new Domain.Models.Account.Account
-                    {
-                        HashedId = "ABC123",
-                        Id = 123,
-                        Name = "Test 1"
-                    }
-                }
-            };
-            Mediator.Setup(x => x.SendAsync(It.IsAny<GetPagedEmployerAccountsQuery>()))
+            ApiService.Setup(s => s.GetAccounts(It.IsAny<string>(), It.IsAny<int>(), It.IsAny<int>(), It.IsAny<CancellationToken>()))
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EAS.Account.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs]: Get rid of IsAny please. 😄

File: src/SFA.DAS.EmployerAccounts.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+using SFA.DAS.EmployerAccounts.TestCommon.Extensions;
+using SFA.DAS.EmployerAccounts.Api.Types;
+
+namespace SFA.DAS.EmployerAccounts.Api.UnitTests.Controllers.EmployerAccountsControllerTests
+{
+    [TestFixture]
+    public class WhenIGetAccounts : EmployerAccountsControllerTests
+    {
+        [Test]
+        public async Task ThenAccountsAreReturnedWithTheirBalanceAndAUriToGetAccountDetails()
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EmployerAccounts.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs]: This API doesn't deal with balances does it?

File: src/SFA.DAS.EAS.Account.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs (hunk 7 above)
Comment (Howesy77) [reply src/SFA.DAS.EAS.Account.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs]: Done

File: src/SFA.DAS.EmployerAccounts.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs (hunk 8 above)
Comment (Howesy77) [reply src/SFA.DAS.EmployerAccounts.Api.UnitTests/Controllers/EmployerAccountsControllerTests/WhenIGetAccounts.cs]: Renamed

**Pull request closed without merging**